src/entities/dataset/build-log.ts

```diff
@@ -14,28 +14,60 @@ export class BuildLog extends BaseEntity {
   @Column({ name: 'type', type: 'enum', enum: Object.values(CubeBuildType), nullable: false })
   type: CubeBuildType;
 
+  @Column({ name: 'user_id', type: 'varchar', nullable: true })
+  userId: string | null;
+
   @CreateDateColumn({ name: 'started_at', type: 'timestamptz', nullable: false })
   startedAt: Date;
 
   @Column({ name: 'completed_at', type: 'timestamptz', nullable: true })
   completedAt: Date | null;
 
+  @Column({ name: 'performance_start', type: 'double precision' })
+  performanceStart: number;
+
+  @Column({ name: 'performance_finish', type: 'double precision', nullable: true })
+  performanceFinish: number | null;
+
+  @Column({ name: 'duration', type: 'double precision', nullable: true })
+  duration: number | null;
+
   @Column({ type: 'text', nullable: true, name: 'build_script' })
   buildScript: string | null;
 
   @Column({ type: 'text', nullable: true, name: 'errors' })
   errors: string | null;
 
-  @ManyToOne(() => Revision, (revision) => revision.builds, { onDelete: 'CASCADE', orphanedRowAction: 'delete' })
+  @ManyToOne(() => Revision, (revision) => revision.builds, {
+    onDelete: 'CASCADE',
+    orphanedRowAction: 'delete',
+    nullable: true
+  })
   @JoinColumn({ name: 'revision_id', foreignKeyConstraintName: 'FK_revision_build_log_id' })
-  revision: Revision;
+  revision: Revision | null;
 
-  public static async startBuild(revision: Revision, type: CubeBuildType, buildId?: string): Promise<BuildLog> {
+  public static async startBuild(
+    revision: Revision | null,
+    type: CubeBuildType,
+    userId?: string,
+    buildId?: string
+  ): Promise<BuildLog> {
     const build = new BuildLog();
+    if (revision) build.revision = revision;
     if (buildId) build.id = buildId;
-    build.revision = revision;
+    if (userId) build.userId = userId;
     build.type = type;
     build.status = CubeBuildStatus.Queued;
+    build.startedAt = new Date();
+    build.performanceStart = performance.now();
```

Comment:
This extends `BaseEntity` which comes with a static constructor helper.

```suggestion
const build = BuildLog.create({
  id,
  revision,
  type,
  userId,
  staus: CubeBuildStatus.Queued,
  startedAt: new Date(),
  performanceStart = performance.now()
});
```

Reply:
Sorry, should've been `id: buildId` but otherwise it works fine. The typing for the object passed to `BaseEntity.create()` is `entityLike: DeepPartial<T>` which means you only need to provide some of the values.

<img width="778" height="491" alt="Image" src="https://github.com/user-attachments/assets/8ad0e3d2-0ba3-4a6b-b9f0-cfdbd2babb96" />
<img width="850" height="168" alt="Image" src="https://github.com/user-attachments/assets/706b008c-ad02-4cbd-b808-f608258875d9" />